app/controllers/users_controller.rb

```diff
@@ -1,26 +1,80 @@
 class UsersController < ApplicationController
+  before_action :find_user, except: [:index, :new, :create]
+  before_action :logged_in_user, except: [:show, :new, :create]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :vefiry_admin, only: :destroy
+
+  def index
+    @users = User.paginate page: params[:page]
+  end
+
   def show
-    @user = User.find params[:id]
   end
 
   def new
     @user = User.new
   end
 
+  def edit
+  end
+
   def create
     @user = User.new user_params
     if @user.save
       log_in @user
       flash[:success] = "Welcome to the Sample App!"
```

Comment:
em nên tìm hiểu thêm về I18n để sửa lại câu thông báo này. I18n in Rails -> keyword